mount/tests/services/servers_test.py

```diff
@@ -20,52 +20,122 @@ async def test_should_not_create_server_with_preexisting_name(ctx: Context):
     )
-    assert data == ServiceError.SERVERS_NAME_ALREADY_EXISTS
+    assert not isinstance(data, ServiceError)
+    assert data["server_name"] == "Akatsuki"
+    assert data["hourly_request_limit"] == 100
+
+    data2 = await servers.create(
+        ctx,
+        server_name="Akatsuki",
+        hourly_request_limit=100,
+    )
+    assert data2 == ServiceError.SERVERS_NAME_ALREADY_EXISTS
 
 
 async def test_should_fetch_one_server(ctx: Context):
-    data = await servers.fetch_one(ctx, server_id=1)
+    data = await servers.create(
+        ctx,
+        server_name="Akatsuki",
+        hourly_request_limit=100,
+    )
     assert not isinstance(data, ServiceError)
     assert data["server_name"] == "Akatsuki"
     assert data["hourly_request_limit"] == 100
 
+    data2 = await servers.fetch_one(ctx, server_name="Akatsuki")
+    assert not isinstance(data2, ServiceError)
+    assert data2["server_name"] == "Akatsuki"
+    assert data2["hourly_request_limit"] == 100
+
 
 async def test_should_not_fetch_one_server(ctx: Context):
-    data = await servers.fetch_one(ctx, server_id=2)
+    data = await servers.fetch_one(ctx, server_id=1)
     assert data == ServiceError.SERVERS_NOT_FOUND
 
 
-async def test_should_fetch_many_servers(ctx: Context):
-    data = await servers.fetch_many(ctx)
-    assert not isinstance(data, ServiceError)
-    assert len(data) == 1
-    assert data[0]["server_name"] == "Akatsuki"
-    assert data[0]["hourly_request_limit"] == 100
+async def test_should_fetch_all_servers(ctx: Context):
+    for server_name in ("Akatsuki", "Ripple", "Gatari"):
+        data = await servers.create(
+            ctx,
+            server_name=server_name,
+            hourly_request_limit=100,
+        )
+        assert not isinstance(data, ServiceError)
+        assert data["server_name"] == server_name
+        assert data["hourly_request_limit"] == 100
+
+    data2 = await servers.fetch_many(ctx)
+    assert not isinstance(data2, ServiceError)
+    assert len(data2) == 3
+
+    for server_data, expected_name in zip(data2, ("Akatsuki", "Ripple", "Gatari")):
+        assert server_data["server_name"] == expected_name
+        assert server_data["hourly_request_limit"] == 100
+
+
+async def test_should_fetch_one_page_of_servers(ctx: Context):
+    for server_name in ("Akatsuki", "Ripple", "Gatari"):
+        data = await servers.create(
+            ctx,
+            server_name=server_name,
+            hourly_request_limit=100,
+        )
+        assert not isinstance(data, ServiceError)
+        assert data["server_name"] == server_name
+        assert data["hourly_request_limit"] == 100
+
+    data2 = await servers.fetch_many(ctx, page=1, page_size=2)
+    assert not isinstance(data2, ServiceError)
+    assert len(data2) == 2
+
+    for server_data, expected_name in zip(data2, ("Akatsuki", "Ripple")):
+        assert server_data["server_name"] == expected_name
+        assert server_data["hourly_request_limit"] == 100
```

Comment:
also includes this new test to improve coverage - tests pagination of servers at the repository layer